app/controllers/incomeSources/add/CheckUKPropertyDetailsController.scala

```diff
@@ -132,46 +154,38 @@ class CheckUKPropertyDetailsController @Inject()(val checkUKPropertyDetails: Che
-          businessDetailsService.createUKProperty(checkUKPropertyViewModel).flatMap {
-            case Left(ex) => Logger("application").error(
-              s"[CheckUKPropertyDetailsController][handleRequest] - Unable to create income source: ${ex.getMessage}")
-              withIncomeSourcesRemovedFromSession(
-                Redirect(redirectErrorUrl)
-              )
-            case Right(CreateIncomeSourceResponse(id)) =>
-              withIncomeSourcesRemovedFromSession(
-                Redirect(getUKPropertyReportingMethodUrl(isAgent, id))
-              ) recover {
-                case _: Exception => Redirect(redirectErrorUrl)
-              }
-          }.recover {
-            case ex: Exception =>
-              if (isAgent) {
-                Logger("application").error(
-                  s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
-                Redirect(redirectErrorUrl)
-              } else {
-                Logger("application").error(
-                  s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
-                Redirect(redirectErrorUrl)
-              }
-          }
-        case Left(ex: Throwable) =>
-          Logger("application").error(
-            s"[CheckUKPropertyDetailsController][handleSubmit] - Error: Unable to build UK property details on submit ${ex.getMessage}")
-          withIncomeSourcesRemovedFromSession(
-            Redirect(redirectErrorUrl)
-          ) recover {
-            case _: Exception => Redirect(redirectErrorUrl)
-          }
-      }
+    getUKPropertyDetailsFromSession(user, ec) flatMap {
+      case Right(checkUKPropertyViewModel: CheckUKPropertyViewModel) =>
+        businessDetailsService.createUKProperty(checkUKPropertyViewModel).flatMap {
+          case Left(ex) => Logger("application").error(
+            s"[CheckUKPropertyDetailsController][handleRequest] - Unable to create income source: ${ex.getMessage}")
+            sessionService.deleteMongoData(JourneyType(Add, UkProperty))
+            Future.successful(Redirect(redirectErrorUrl))
+
+          case Right(CreateIncomeSourceResponse(id)) =>
+            sessionService.deleteMongoData(JourneyType(Add, UkProperty))
+            Future.successful(Redirect(getUKPropertyReportingMethodUrl(isAgent, id)))
+        }.recover {
+          case ex: Exception =>
+            if (isAgent) {
+              Logger("application").error(
+                s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
+              Redirect(redirectErrorUrl)
+            } else {
+              Logger("application").error(
+                s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
+              Redirect(redirectErrorUrl)
```

Comment:
Should fix: duplicated code here, I'm not sure we need different logging for agents. Even so, it could be done in 1 line with a dynamic error message